app/lib/Plugins/SearchEngine/Elastic8/Query.php

```diff
@@ -435,6 +436,20 @@ protected function getFilterQuery(): string {
 		return join(' AND ', $terms);
 	}
 
+	protected function alterFilter($filter) {
```

Comment:
I hate everything in this function.

Reply:
We should be able to use `$this->getFieldTypeForTerm($first_term)` here as well